### PR TITLE
Experimental CI changes

### PR DESCRIPTION
**Pull request by aMahanna**
Branched from #291 

Experimenting with Github Actions & CircleCI to make use of matrices to cover a variety of testing environments

```py
              python_version: ["3.10"] # ["3.8", "3.9", "3.10", "3.11", "3.12"]
              arangodb_config: ["single"] # ["single", "cluster"]
              arangodb_license: ["community"] # ["community", "enterprise"]
              arangodb_version: ["latest"] # ["3.10.10", "3.11.4", "latest"]
```

1. Updates `.circleci/config.yml`
2. Renames `build.yaml` to `docs.yaml`
3. Updates `starter.sh`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,164 +1,113 @@
...
-          arangodb-version: "enterprise-preview:latest"
-          arangodb-config: "cluster.conf"
-          cluster: true
-          enterprise: true
+      # - run:
+      #     name: Upload to Coveralls
+      #     command: |
+      #       if [ "<< parameters.python_version >>" = "3.11" && "<< parameters.arangodb_config >>" = "single" && "<< parameters.arangodb_license >>" = "community" && "<< parameters.arangodb_version >>" = "latest"  ]; then
+      #         coveralls/upload
+      #       fi
```
Comment (aMahanna) [inline .circleci/config.yml:113]: Uploading coverage won't be possible until we request access from the Organization Admin:

image

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,164 +1,113 @@
...
-          - >
-            while ! wget -q -O /dev/null http://localhost:8000/$PROJECT/tests/static/setup.sh; do sleep 1; done &&
-            wget -O - http://localhost:8000/$PROJECT/tests/static/setup.sh |
-            /bin/sh
+      arangodb_version:
+        type: string
+    # TODO: Reconsider using a docker image instead of a machine
+    # i.e cimg/python:<< parameters.python_version >>
+    machine:
+      image: ubuntu-2204:current
```
Comment (aMahanna) [inline .circleci/config.yml:58]: Hopefully we can revisit using `cimg/python` with `starter.sh` as that would cut the job times from ~4 minutes to ~2 minutes

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,164 +1,113 @@
...
+
+workflows:
+  ci:
+    jobs:
+      - lint
+      - test:
+          matrix:
+            parameters:
+              # TODO: Revisit why pyenv doesn't recognize 3.12
+              python_version: ["3.8", "3.9", "3.10", "3.11"] # "3.12"
```
Comment (aMahanna) [inline .circleci/config.yml:14]: For some reason `pyenv 2.3.37` does not recognize `3.12` when attempting to do `pyenv install 3.12`...

However, `pyenv install 3.12` worked fine locally with `pyenv 2.3.31` 🤔

File: .github/workflows/docs.yaml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    inputs:
+      debug_enabled:
+        type: boolean
+        description: Debug with tmate
+        required: false
+        default: false
+
+jobs:
+  # This has been migrated to CircleCI
+  # test:
```
Comment (aMahanna) [inline .github/workflows/docs.yaml:15]: Leaving the `test` job as a reference should we ever need to move back to testing on Github Actions

Average job time here is ~1-2 minutes, but the disadvantage is that we lose access to CircleCI's built-in analytics dashboards

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,164 +1,113 @@
...
+              arangodb_config: ["single", "cluster"]
+              arangodb_license: ["community", "enterprise"]
```
Comment (aMahanna) [inline .circleci/config.yml:16]: Want thing I just want to clarify according to the current state of `starter.sh`

`community` --> `docker/arangodb` (https://hub.docker.com/r/arangodb/arangodb/tags)
`enterprise` --> `docker/enterprise` (https://hub.docker.com/r/arangodb/enterprise/tags)

Perhaps we can revisit the idea of `community-preview` (https://hub.docker.com/r/arangodb/arangodb-preview/tags) and `enterprise-preview` (https://hub.docker.com/r/arangodb/enterprise-preview/tags)

**Review by apetenchea — APPROVED** [review]
LGTM

**Pull request merged**